src/block/samples.rs

```diff
@@ -122,31 +124,77 @@ pub trait FromNativeSample: Sized + Copy + Default + 'static {
 
     /// Create this sample from a u32, trying to represent the same numerical value
     fn from_u32(value: u32) -> Self;
+
+    /// Convert all values from the slice into this type.
+    /// This function exists to allow the compiler to perform a vectorization optimization.
+    #[inline]
+    fn from_f16s(from: &[f16], to: &mut [Self]) {
```

Comment:
Auto-vectorization is not going to happen for `f16`. This should recommend overriding this method instead of referencing auto-vectorization.

Reply:
It is in fact overridden in the implementation. But sure, we can hint to that in this part of the documentation too.